repose-aggregator/components/filters/valkyrie-authorization/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala

```diff
@@ -135,6 +137,42 @@ class ValkyrieAuthorizationFilter @Inject()(configurationService: ConfigurationS
       }
     }
 
+    def getInventory(headerResult: ValkyrieResult): ValkyrieResult = {
+      def parseInventory(inputStream: InputStream): Try[UserPermissions] = {
+
+        @tailrec
+        def parseJson(deviceToPermissions: List[DeviceToPermission], values: List[JsValue]): UserPermissions = {
+          if (values.isEmpty) {
+            UserPermissions(List(ACCOUNT_ADMIN).toVector, deviceToPermissions.toVector)
```

Comment:
Later on we add these to the existing UserPermissions which means the "account_admin" role will show up twice now.  This should be Vector.empty.  Or, better yet, have the parseJson method just return a Vector of DeviceToPermissions instead of UserPermissions.  Up to you.